src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -691,6 +681,10 @@ public ImmutableArray<T> Replace(T oldValue, T newValue)
         public ImmutableArray<T> Replace(T oldValue, T newValue, IEqualityComparer<T> equalityComparer)
         {
             var self = this;
+            
+            if (self.IsEmpty)
```

Comment:
:exclamation: This changes the semantics of the method. Previously, if `self` is empty the method always threw an exception.